src/LuaTransformer.ts

```diff
@@ -3033,7 +3113,16 @@ export class LuaTransformer {
             || this.currentNamespace
             || (tsOriginal && tsHelper.findFirstNodeAbove(tsOriginal, ts.isFunctionLike))
         ) {
-            statements.push(tstl.createVariableDeclarationStatement(lhs, undefined, parent));
+            if (!isArray(lhs)) {
+                lhs = [lhs];
+            }
+            const shouldExport = lhs.some(i => this.isIdentifierExported(i.text));
+            if (shouldExport) {
+                statements.push(
+                    tstl.createAssignmentStatement(lhs.map(i => this.createExportedIdentifier(i)), undefined, parent));
```

Comment:
This looks strange, if there is one thing that should be exported, it calls createdExportedIdentifier for all of them? Does that mean createExportedIdentifier handles non-exported stuff too? Why not just always call this then? What is the point of shouldExport?

Reply:
`export let x, y, z;` will result in all 3 identifiers being exported.